### PR TITLE
feat: add tensorflow keras classification pipeline

### PR DESCRIPTION
**Pull request by NeethuES-intel**
## PR Checklist

- [X] Added label to the Pull Request for easier discoverability and search
- [ ] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/retail-use-cases/blob/main/CONTRIBUTING.md
- [ ] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
Added tensorflow keras classification pipeline with performance evaluation feature

## Issue this PR will close

close: #**52**

## Anything the reviewer should know when reviewing this PR?
Model is not detecting all objects, experimenting with other models. Will replace if I find any better that is available in keras.

## Test Instructions if applicable

- Make sure input source video is added to performance-tools/sample-media folder and update path in [tensorflow-keras-env.txt](https://github.com/intel-retail/retail-use-cases/compare/main...NeethuES-intel:retail-use-cases:tensorflow-integration?expand=1#diff-67158b975325d077d52e85c421b67051d3471dc0cbb77d22ce6765f4711a00ac)
- make build_tensorflow_keras
- make run_tensorflow_keras_classification
- run 2 pipelines with PIPELINE_COUNT=2 make run_tensorflow_keras_classification
- make down_tensorflow_keras_classification
- results of the detection (only object detected) is published to mqtt broker (topic - tensorflow_keras_classification_results) and logged into results.txt, fps in pipelines.txt

### CODE REVIEW

**Review by seanohair22 — COMMENTED** [review]
Built correctly, was able to run demo, it did crash after ~10 seconds, I did not see any results files nor mqtt output

File: use-cases/demos/classification/tensorflow-keras/Tesnorflow_Keras_Example.ipynb [hunk 1]
```diff
@@ -0,0 +1,957 @@
+{
```
Comment (antoniomtz) [inline use-cases/demos/classification/tensorflow-keras/Tesnorflow_Keras_Example.ipynb:1]: is this file needed?

**Review by antoniomtz — CHANGES_REQUESTED** [review]

**Comment by NeethuES-intel** [comment]
> Built correctly, was able to run demo, it did crash after ~10 seconds, I did not see any results files nor mqtt output

Do you have any docker logs or screenshots ? Did all containers fail ?
It does run slow.

File: use-cases/demos/classification/tensorflow-keras/Tesnorflow_Keras_Example.ipynb (hunk 1 above)
Comment (NeethuES-intel) [reply use-cases/demos/classification/tensorflow-keras/Tesnorflow_Keras_Example.ipynb:1]: > is this file needed?

Notebook is for easy and quick visualization and testing. I added it for pytorch so kept it for tensorflow as well.

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
built and ran- the model gives the pop_bottle classification until almost towards to the end, at which it doesn't give the classifications for the two vague coke bottles.

also i have some feedback comments for the code.

**Comment by NeethuES-intel** [comment]
> built and ran- the model gives the pop_bottle classification until almost towards to the end, at which it doesn't give the classifications for the two vague coke bottles.
> 
> also i have some feedback comments for the code.

Unfortunately this is a model issue. I am experimenting with other models available through Keras CV. I have addressed your other 2 comments.

**Comment by NeethuES-intel** [comment]
> > > built and ran- the model gives the pop_bottle classification until almost towards to the end, at which it doesn't give the classifications for the two vague coke bottles.
> > > also i have some feedback comments for the code.
> > 
> > 
> > Unfortunately this is a model issue. I am experimenting with other models available through Keras CV. I have addressed your other 2 comments.
> 
> ok, do they offer some threshold to adjust for that, or that is pretty much built-in in the model itself?

Yes, I am capping the threshold to 0.5 in the code. Seeing lot of false positives if I don't do that.

**Review by antoniomtz — APPROVED** [review]

**Comment by seanohair22** [comment]
> > Built correctly, was able to run demo, it did crash after ~10 seconds, I did not see any results files nor mqtt output
> 
> Do you have any docker logs or screenshots ? Did all containers fail ? It does run slow.

Only the tensorflow container failed, the mqtt container remained up. I was trying with a single pipeline only.
Yes, here is the docker log from the tensorflow container: 
[docker_tensorflow_log.txt](https://github.com/user-attachments/files/17782596/docker_tensorflow_log.txt)

**Comment by NeethuES-intel** [comment]
> > > Built correctly, was able to run demo, it did crash after ~10 seconds, I did not see any results files nor mqtt output
> > 
> > 
> > Do you have any docker logs or screenshots ? Did all containers fail ? It does run slow.
> 
> Only the tensorflow container failed, the mqtt container remained up. I was trying with a single pipeline only. Yes, here is the docker log from the tensorflow container: [docker_tensorflow_log.txt](https://github.com/user-attachments/files/17782596/docker_tensorflow_log.txt)

Thanks Sean. Error looks interesting, haven't seen this before. Did the demo video open up for you ? May be some hardware specific issue. Will sync offline on this issue.

**Review by seanohair22 — APPROVED** [review]
Retested, LGTM!
